radix-engine/src/model/transaction_processor/executables.rs

```diff
@@ -80,14 +80,30 @@ impl<'a> NativeInvocation for TransactionProcessorRunInvocation<'a> {
             match instruction {
                 Instruction::CallFunction { args, .. }
                 | Instruction::CallMethod { args, .. }
-                | Instruction::CallNativeFunction { args, .. }
-                | Instruction::CallNativeMethod { args, .. } => {
+                | Instruction::CallNativeFunction { args, .. } => {
                     let scrypto_value =
                         IndexedScryptoValue::from_slice(&args).expect("Invalid CALL arguments");
                     for global_address in scrypto_value.global_references() {
                         node_refs_to_copy.insert(RENodeId::Global(global_address));
                     }
                 }
+                Instruction::CallNativeMethod { args, method_ident } => {
+                    let scrypto_value =
+                        IndexedScryptoValue::from_slice(&args).expect("Invalid CALL arguments");
+                    for global_address in scrypto_value.global_references() {
+                        node_refs_to_copy.insert(RENodeId::Global(global_address));
+                    }
+
+                    // TODO: This needs to be cleaned up
+                    // TODO: How does this relate to newly created vaults in the transaction frame?
+                    // TODO: Will probably want different spacing for refed vs. owned nodes
```

Comment:
I thought we agreed to limit receiver type to `&mut self` only. Owned objects need another layer of indirection for addressing, like buckets and proofs, subject to product agreement.

Reply:
it is `&mut self` only. I'm talking about refed vs. owned in caller frame NOT whether the function is consuming or non-consuming.